src/_pytest/debugging.py

```diff
@@ -275,12 +275,14 @@ def _test_pytest_function(pyfuncitem):
     _pdb = pytestPDB._init_pdb("runcall")
     testfunction = pyfuncitem.obj
     pyfuncitem.obj = _pdb.runcall
-    if "func" in pyfuncitem._fixtureinfo.argnames:  # pragma: no branch
+    if "func" in pyfuncitem.funcargs:  # pragma: no branch
         raise ValueError("--trace can't be used with a fixture named func!")
     pyfuncitem.funcargs["func"] = testfunction
-    new_list = list(pyfuncitem._fixtureinfo.argnames)
-    new_list.append("func")
-    pyfuncitem._fixtureinfo.argnames = tuple(new_list)
+    if "func" not in pyfuncitem._fixtureinfo.argnames:
```

Comment:
indeed this workaround requires a comment

practically this belongs to the FunctionDefininition (and should only happen once at that)
unfortunately with the current design that's not exposed and we have a hack instead

the ""correct" fix is not to have more than one place where this is added, but that's for after FunctionDefinition lands